fs/vfs/fs_poll.c

```diff
@@ -284,23 +285,27 @@ void poll_notify(FAR struct pollfd **afds, int nfds, pollevent_t eventset)
       fds = afds[i];
       if (fds != NULL)
         {
-          /* race condition protection when modifying fds->revents */
-
-          flags = enter_critical_section();
-
           /* The error event must be set in fds->revents */
 
-          fds->revents |= eventset & (fds->events | POLLERR | POLLHUP);
-          if ((fds->revents & (POLLERR | POLLHUP)) != 0)
+          revents = eventset & (fds->events | POLLERR | POLLHUP);
+          if (revents != 0)
             {
-              /* Error or Hung up, clear POLLOUT event */
+              /* race condition protection when modifying fds->revents */
 
-              fds->revents &= ~POLLOUT;
-            }
+              flags = enter_critical_section();
+
+              fds->revents |= revents;
+              if ((fds->revents & (POLLERR | POLLHUP)) != 0)
+                {
+                  /* Error or Hung up, clear POLLOUT event */
+
+                  fds->revents &= ~POLLOUT;
+                }
 
-          leave_critical_section(flags);
+              leave_critical_section(flags);
+            }
 
-          if ((fds->revents != 0 || (fds->events & POLLALWAYS) != 0) &&
+          if (revents != 0 || (fds->events & POLLALWAYS) != 0 &&
```

Comment:
This is not equivalent to the original condition, now if fds->cb = NULL you get a crash